data_structures/linked_list/swap_nodes.py

```diff
@@ -11,8 +11,9 @@ def __init__(self):
     def print_list(self):
         temp = self.head
         while temp is not None:
-            print(temp.data)
+            print(temp.data, end=' ')
             temp = temp.next
+        print()
 
     # adding nodes
     def push(self, new_data):

```

Comment:
Needs type hints and more pythonic condition checking

Reply:
Do we want to limit the data type?  Currently this can be used with many different data types and I consider that a feature, not a bug.

Reply:
Then let's be consistent by using `typing.Any`